ykcompile/src/lib.rs

```diff
@@ -225,65 +261,158 @@ impl<TT> TraceCompiler<TT> {
         Ok(())
     }
 
+    /// Get the type of a place.
+    fn place_ty(&self, p: &Place) -> &Ty {
+        let base_ty = SIR.ty(&self.local_decls[&p.local].ty);
+        match p {
```

Comment:
I found this unnecessarily hard to read. Can we turn this into a normal if block, e.g.
```
if let Place { local, projection } = p {
    if projection.is_empty() {
        base_ty
    } else {
        ...
    }
}
```

Reply:
Sure